dynatrace/src/main/java/com/dynatrace/opentelemetry/metric/DynatraceMetricExporter.java

```diff
@@ -200,13 +201,18 @@ protected CompletableResultCode export(
                   new InputStreamReader(connection.getInputStream(), Charsets.UTF_8));
           resultCode = handleSuccess(code, metricLines.size(), response);
         } else {
-          String response =
-              CharStreams.toString(
-                  new InputStreamReader(connection.getErrorStream(), Charsets.UTF_8));
-          logger.warning(
-              () ->
+          if (logger.isLoggable(Level.WARNING)) {
+            InputStream errorStream = connection.getErrorStream();
+            if (errorStream != null) {
+              String message =
+                  CharStreams.toString(new InputStreamReader(errorStream, Charsets.UTF_8));
+              logger.warning(
                   String.format(
-                      "Error while exporting. Status code: %d; Response: %s", code, response));
+                      "Error while exporting. Status code: %d; Response: %s", code, message));
+            } else {
+              logger.warning(String.format("Error while exporting. Status code: %d", code));
```

Comment:
I guess we could use the `() =>` approach for the logger because that does not allocate the string right away.

Reply:
Also now there's quite more logic here.. I wonder if it makes sense to put this into a separated method? 🤔